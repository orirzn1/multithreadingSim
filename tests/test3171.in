L15
S8
O9
N7

T0
I@0x00000000
STORE $0, $6, 0x3
STORE $1, $4, 0x3
STORE $4, $7, 0x3
STORE $6, $5, 0x3
STORE $5, $2, $7
STORE $5, $3, $0
HALT $0

T1
I@0x00000000
LOAD $7, $6, 0x4
SUBI $0, $3, 0x1
SUBI $0, $1, 0x1
SUBI $4, $4, 0x2
SUBI $4, $7, 0x3
HALT $0

T2
I@0x00000000
ADDI $7, $6, 0x2
LOAD $7, $4, 0x0
ADD $2, $1, $0
SUB $0, $5, $5
ADDI $2, $5, 0x4
SUB $2, $4, $6
SUBI $7, $7, 0x2
STORE $2, $3, $5
SUBI $4, $3, 0x4
STORE $1, $0, 0x3
HALT $0

T3
I@0x00000000
ADD $7, $3, $3
SUB $0, $6, $1
SUBI $1, $0, 0x2
LOAD $3, $4, 0x4
ADD $2, $4, $2
STORE $1, $3, $4
STORE $6, $0, 0x4
SUB $5, $4, $4
LOAD $4, $7, $4
HALT $0

T4
I@0x00000000
LOAD $5, $7, $1
SUB $0, $5, $7
ADD $2, $4, $3
LOAD $5, $3, 0x2
ADDI $1, $4, 0x3
SUB $1, $5, $6
ADDI $0, $7, 0x3
ADD $2, $1, $5
SUBI $6, $4, 0x4
LOAD $1, $5, $7
HALT $0

T5
I@0x00000000
STORE $2, $4, $4
STORE $3, $4, $6
SUB $5, $4, $1
STORE $7, $2, 0x2
LOAD $1, $5, 0x0
SUBI $5, $1, 0x4
SUB $3, $6, $0
LOAD $4, $4, 0x0
HALT $0

T6
I@0x00000000
LOAD $5, $1, $5
ADDI $4, $1, 0x1
SUBI $1, $0, 0x2
ADD $7, $0, $6
ADDI $5, $3, 0x1
HALT $0

D@0x0
0x7
0x9
0x5
0x8
0x3
