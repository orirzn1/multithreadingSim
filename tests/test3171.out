
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x6	R5 = 0x0	R6 = 0x0	R7 = 0x9
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 3:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x9	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x3	R1 = 0x5	R2 = 0x12	R3 = 0x0	R4 = 0x0	R5 = 0x9	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x1	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 5.433333

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 3:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x3	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x1	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.250000

